FamliCareClientProject/famlicare-client-project
# Revert to 831254f

## Description
This pull request addresses the issue where the app launches but the page clocks and never loads, potentially due to an infinite loop. The changes implement safeguards against rapid state changes, improve error handling, and add more detailed logging to help diagnose and prevent such issues in the future.

## Changes Made
1. **NavigationHandler.js**:
   - Implemented a debounce mechanism to prevent rapid successive navigation attempts.
   - Added more detailed logging for navigation actions.

2. **user.saga.js**:
   - Enhanced `fetchUserSaga` and `refreshUserSaga` with more detailed logging.
   - Improved handling of invalid or empty user data.
   - Added proper error handling and redirection to the login page on authentication failures.

## How to Test
1. Launch the application and verify that it loads properly without getting stuck in an infinite loop.
2. Test various navigation scenarios, especially those involving user authentication.
3. Check the console logs for more detailed information about user fetching and navigation actions.
4. Verify that the application properly redirects to the login page when authentication fails or user data is invalid.

## Additional Notes
- Monitor the application logs carefully after implementing these changes to see if the infinite loop issue is resolved.
- If the issue persists, further investigation may be needed in areas such as server-side authentication middleware, Redux store configuration, or other sagas/middleware that might interfere with the user authentication flow.
